src/ui/inventory.ts

```diff
@@ -137,22 +153,176 @@ export class Inventory extends Base {
 
     const size = ITEM_SIZE[record.size];
 
-    // Temporarily remove this item from the positions array to avoid false overlap
+    // Temporarily remove this item from positions to avoid self-collision
     const otherPositions = this.positions.filter((p) => p.id !== itemId);
-
-    // Reuse your `doesItemFitAt` function but pass in the reduced list
     const fits = this.doesItemFitAt(position.tab, x, y, size, otherPositions);
     if (!fits) return;
 
-    // Update position
     position.x = x;
     position.y = y;
-
-    // Re-render
     this.render();
     this.savePositions();
   }
 
+  private startDrag(itemId: number, element: HTMLElement, e: PointerEvent) {
+    if (this.dragState.active) return;
+
+    playSfxById(SfxId.InventoryPickup);
+
+    this.dragState.active = true;
+    this.dragState.itemId = itemId;
+    this.dragState.element = element;
+    this.dragState.pointerId = e.pointerId;
+
+    // Only grab the image - don't want the tooltip following our cursor
+    const img = element.querySelector('img') as HTMLImageElement;
+    const clone = img.cloneNode(true) as HTMLImageElement;
+    clone.style.position = 'fixed';
+    clone.style.pointerEvents = 'none';
+    clone.style.zIndex = '9999';
+    clone.style.opacity = '0.8';
+    clone.classList.add('dragging-clone');
+
+    const rect = element.getBoundingClientRect();
+    this.dragState.offset = {
+      x: e.clientX - rect.left,
+      y: e.clientY - rect.top,
+    };
+
+    clone.style.left = `${e.clientX - this.dragState.offset.x}px`;
+    clone.style.top = `${e.clientY - this.dragState.offset.y}px`;
+
+    document.body.appendChild(clone);
+    this.dragState.clone = clone;
+
+    // Visual feedback for original item
+    element.style.opacity = '0.5';
+    element.style.cursor = 'grabbing';
+
+    // These pointer events won't mess with keyboard like the old drag system did
+    document.addEventListener('pointermove', this.handleDragMove);
+    document.addEventListener('pointerup', this.handleDragEnd);
+    document.addEventListener('pointercancel', this.handleDragEnd);
+
+    element.setPointerCapture(e.pointerId);
+  }
+
+  private handleDragMove(e: PointerEvent) {
+    if (
+      !this.dragState.active ||
+      e.pointerId !== this.dragState.pointerId ||
+      !this.dragState.clone
+    ) {
+      return;
+    }
+
+    // Update visual clone position
+    this.dragState.clone.style.left = `${e.clientX - this.dragState.offset.x}px`;
+    this.dragState.clone.style.top = `${e.clientY - this.dragState.offset.y}px`;
+
+    this.updateDropZoneHighlights(e.clientX, e.clientY);
```

Comment:
I don't think this is working. I don't see the inventory highlighted when dragging.